contents/javascript/VueJs/Intro.md

```diff
@@ -0,0 +1,202 @@
+- [What is VueJs and What are its Advantages](https://hackernoon.com/what-is-vue-js-and-what-are-its-advantages-4071b7c7993d)
+- [Pros and Cons of the Vue.js framework - Vue.js Feed](https://vuejsfeed.com/blog/pros-and-cons-of-the-vue-js-framework)
+
+Detailed Comparison of VueJs with other JavaScript frameworks can be found from:
+- [Vue Guild: Comparison with Other Frameworks](https://vuejs.org/v2/guide/comparison.html)
+- [Angular vs React vs Vue](https://medium.com/unicorn-supplies/angular-vs-react-vs-vue-a-2017-comparison-c5c52d620176)
+
+## Drawbacks of VueJS
+
+#### Relatively small size community
+
+VueJs is a relatively new JavaScript framework as compared to Augular and React. The size of the community for VueJs is therefore relatively small. Although small size community means you can differentiate yourself from other JavaScript developers, it also means there are fewer resources such as tutorials and problem-shooting guides.
+
+#### Language barriers
+
+A majority of users of VueJs are the Chinese as VueJs is developed by a Chinese American. He is supportive of the Chinese community and hence a lot of the existing plugins are written in Chinese. There might be some language barriers for an English speaking developer seeking for VueJs resources.
+
+## Installation
+
+There are many ways to install VueJs. Some of the following ways are mentioned below:
+* Import VueJs in HTML `<script>` tag:
+  - [development version](https://vuejs.org/js/vue.js)
+  - [product version](https://vuejs.org/js/vue.min.js)
+* CDN:
+  - include the following into the html file:
+    `<script src="https://cdn.jsdelivr.net/npm/vue@2.5.13/dist/vue.js"</script>`
+* NPM:
+```npm
+    $ npm install vue
+```
+* CLI:
+```cli
+    $ npm install --global vue-cli  # install vue-cli
+    $ vue init webpack my-project  # create a new project using the "webpack" template
+    $ cd my-project  # install dependencies and go!
+    $ npm install
+    $ npm run dev
+```
+
+Apart from installing VueJs, you can install [VueJs development tools](https://github.com/vuejs/vue-devtools#vue-devtools) in your browser. This will allow you to inspect and debug the Vue components in your projects.
+
+## Creating a Simple Project in VueJs:
+
+Steps to print the VueJs version of "Hello World":
+
+* Import VueJs CDN and the JavaScript file in the main HTML file.
+```HTML
+  <script src="https://unpkg.com/vue@2.5.13/dist/vue.js"></script>
+  <script src="the_path_to_the_javacript_file.js"></script>
+```
+
+* Create a Vue instance in the JavaScript file; bind the instance to one of the component in our html file (the `root` element). We can only use the Vue data inside the component with id `root` as we bind it to the Vue element.
+  - In this case, only this component is working in VueJs while the rest are unaffected. This is how we progressively plug in VueJs into our projects without a complete one at a go.
```

Comment:
Which component is `this component` (try `only the root component can be accessed in VueJs` or something similar? I'm not sure what you're trying to say here that is different from `We can only use the Vue data inside the component with id `root` as we bind it to the Vue element`.
Also `progressively plug VueJs into our projects` is enough, not sure what the second part of the sentence means.